File: src/main/java/org/example/Domain/Inscriere.java
package org.example.Domain;


public class Inscriere extends Entity <Pair<Long,Long>>{
    Cursa cursa;
    Participant participant;

    public Inscriere(Pair<Long,Long> id,Cursa cursa, Participant participant) {
        super.id = id;
        this.cursa = cursa;
        this.participant = participant;
    }

    public Cursa getEchipa() {
        return cursa;
    }

    public void setEchipa(Cursa echipa) {
        this.cursa = echipa;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    @Override
    public String toString() {
        return "Inscriere{" +
                "cursa=" + cursa +
                ", participant=" + participant +
                '}';
    }
}

File: Domain/Inscriere.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Motociclete.Domain
{
    public class Inscriere : Entity<Pair<long,long>>
    {
        public Participant Participant {  get; set; }
        public Cursa Cursa { get; set; }

        public Inscriere(long id,Participant participant, Cursa cursa)
        {
            this.id = new Pair<long, long>(participant.id, cursa.id);
            this.Participant = participant;
            this.Cursa = cursa;
        }
    }
}
